tests/test_scalar_bars.py

```diff
@@ -54,3 +54,15 @@ def test_clear(scalar_bars):
     assert len(scalar_bars) == 1
     scalar_bars.clear()
     assert len(scalar_bars) == 0
+
+
+def test_actor_removal(sphere):
+    # verify that when removing an actor we also remove the
+    # corresponding scalar bar
+
+    sphere['scalars'] = sphere.points[:, 2]
+
+    pl = pv.Plotter()
+    actor = pl.add_mesh(sphere, show_scalar_bar=True)
+    pl.remove_actor(actor)
```

Comment:
I'm probably paranoid but I'd also check the count before the test (so that we ensure that the scalar bar was added first):
```suggestion
    assert len(pl.scalar_bars) == 1
    pl.remove_actor(actor)
```

Reply:
Agreed.